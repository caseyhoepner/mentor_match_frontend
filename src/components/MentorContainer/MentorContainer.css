.mc-header {
  background-color: grey;
}
.mc-title-text {
  margin-bottom: 20px;
}
.mc-nav-btns {
  align-items: center;
  display: flex;
  justify-content: space-between;
  padding: 20px 40px;
}
.mc-github-btn {
  align-items: center;
  color: #fff;
  display: flex;
  font-weight: 300;
}
.mc-gh-tagline {
  letter-spacing: 1px;
}
.mc-github-logo {
  height: 40px;
  margin-left: 20px;
  width: 40px;
}
.mc-turing-logo {
  height: 100px;
}
.mc-title {
  align-items: center;
  color: #fff;
  display: flex;
  flex-direction: column;
  font-size: 1.75rem;
  font-weight: 300;
  padding: 10px 0 70px;
}
.mc-line-break {
  background-color: #03c1d1; 
  height: 1px;
  width: 300px;
}
.mc-tagline {
  letter-spacing: 1px;
  font-size: 1.5rem;
}

.mc-card-container {
  display: flex;
  flex-wrap: wrap;
  justify-content: flex-start;
  width: 90%;
  margin: auto;
}
