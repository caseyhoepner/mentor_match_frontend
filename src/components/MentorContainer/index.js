import React, { Component } from 'react';
import './MentorContainer.css';
import Search from '../../containers/Search';
import { MentorCard } from '../../containers/MentorCard';
import { connect } from 'react-redux';
import uuid from 'uuid';

export class MentorContainer extends Component {

  filterBySearchTerm = () => {
    let { searchTerm, mentors } = this.props

    const searchedMentors = mentors.reduce((acc, mentor) => {
      const mentorKeys = Object.keys(mentor)
      const foundMentor = mentorKeys.find((key) => {
        let newMentor;
        if (typeof mentor[key] === 'string') {
          newMentor = (mentor[key].toLowerCase()).includes(searchTerm.toLowerCase())
        }
        return newMentor
      })
      if (foundMentor) {
        acc.push(mentor)
      }
      return acc
    },[])

    return searchedMentors
  }

  filterByLocale = (cards) => {
    let { locale } = this.props
    let filteredCards;
    
    if (locale === 'Denver') {
      filteredCards = cards.filter((card) => !card.props.mentor.meeting_location.includes('Remote'))
    } else if (locale === 'Remote') {
      filteredCards = cards.filter((card) => card.props.mentor.meeting_location.includes('Remote'))
    }
    return filteredCards
  }

  filterByPreference = (studentCards) => {
    let { mentorFilters } = this.props
    let filterKeys = Object.keys(mentorFilters)
    let finalCards = []

    let trueVals = filterKeys.filter((key) => {
      let foundKey;
      if (mentorFilters[key] === true) {
        foundKey = key
      }
      return foundKey
    });

    studentCards.forEach((card) => {   
      let checkedCard = this.checkPrefs(card, trueVals)
      if (checkedCard) {
        finalCards.push(checkedCard)
      }
    })    
    return finalCards
  }

  cleanPrefs = (str) => {
    if (str === 'Front-End') {
      return 'frontEnd'
    } else if (str === 'Back-End') {
      return 'backEnd'
    } else {
      return str.toLowerCase()
    }
  }

  checkPrefs(card, valArr) {
    let { identity_preference, stack_preference } = card.props.mentor
    let totalPrefs = [...identity_preference, stack_preference]
    totalPrefs = totalPrefs.map((pref) => {
      return this.cleanPrefs(pref)
    })
    let finalCard;

    if (valArr.length === 2) {
   
      if (totalPrefs.includes(valArr[0]) 
        && totalPrefs.includes(valArr[1])
      ) {
        finalCard = card
      }

    } else if (valArr.length === 1) {

      if (totalPrefs.includes(valArr[0])) {
        finalCard = card
      }

    }
    return finalCard
  }

  makeMentorCards = (mentorCards) => {
    return mentorCards.map(mentor => {
      return <MentorCard key={uuid()} mentor={mentor} />
    })
  }

  render() {
    let { mentors, showingAllMentors, searchTerm, mentorFilters, locale } = this.props
    let mentorCards;

    if (!showingAllMentors) {
      const filteredMentors = mentors.filter(mentor => {
        return mentor.matched === false;
      })
        mentorCards = filteredMentors.map(mentor => {
        return <MentorCard key={uuid()} mentor={mentor}/>
      })
    } else {
        mentorCards = mentors.map(mentor => {
        return <MentorCard key={uuid()} mentor={mentor}/>
      })
    }

    if (searchTerm) {
      const searchedMentors = this.filterBySearchTerm()
      mentorCards = searchedMentors.map(mentor => {
        return <MentorCard key={uuid()} mentor={mentor}/>
      })
    }

    if (locale) {
      mentorCards = this.filterByLocale(mentorCards)
    }

    if (Object.values(mentorFilters).includes(true)) {
      mentorCards = this.filterByPreference(mentorCards);
    }

    mentorCards = this.makeMentorCards(mentorCards);

    return (
      <div>
        <header className='mc-header'>
          <div className='mc-nav-btns'>
            <img src={require('../../utils/assets/turing-full-black.png')} alt='Turing Logo' className='mc-turing-logo' />
            <div className='mc-github-btn'>
              <a href='https://github.com/login/oauth/authorize?client_id=81ef33f7e5b1e21f0162&scope=repo' className='mc-gh-tagline'>Sign in with GitHub</a>
              <img src={require('../../utils/assets/github-logo.svg')} alt='GitHub Logo' className='mc-github-logo' />
            </div>
          </div>
          <div className='mc-title'>
            <h1 className='mc-title-text'>Mentor Profiles</h1>
            <div className='mc-line-break'></div>
            <p className='mc-tagline'>Browse current mentors to find your match!</p>
          </div>
        </header>
        <Search />
        <div className='mc-card-container'>
          { mentorCards }
        </div>
      </div>
    )
  }
}

export const mapStateToProps = (state) => ({
  mentors: state.mentors,
  mentorFilters: state.mentorFilters,
  locale: state.locale,
  searchTerm: state.searchTerm,
  showingAllMentors: state.showingAllMentors
})

export default connect(mapStateToProps)(MentorContainer);