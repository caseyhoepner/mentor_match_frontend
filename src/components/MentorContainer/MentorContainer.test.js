import React from 'react';
import { shallow } from 'enzyme';
import MentorContainer from './';
import { BrowserRouter } from 'react-router-dom';

describe('MentorContainer', () => {
  let wrapper;
  
  beforeEach(() => {
    wrapper = shallow(<MentorContainer />)
  });

  it('matches the snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });
});