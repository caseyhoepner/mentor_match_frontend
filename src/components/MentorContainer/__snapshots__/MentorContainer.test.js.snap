// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MentorContainer matches the snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <MentorContainer
    mentorFilters={
      Object {
        "backEnd": false,
        "female": false,
        "frontEnd": true,
        "lgbtq": false,
        "male": false,
        "parent": false,
        "veteran": false,
      }
    }
    mentors={
      Array [
        Object {
          "identity_preference": Array [
            "female-identifying",
          ],
          "meeting_location": "Remote",
          "name": "Casey",
          "stack_preference": "Front-End",
        },
        Object {
          "identity_preference": Array [
            "male-identifying",
          ],
          "meeting_location": "Denver",
          "name": "Alex",
          "stack_preference": "Back-End",
        },
      ]
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <header
          className="mc-header"
        >
          <div
            className="mc-nav-btns"
          >
            <img
              alt="Turing Logo"
              className="mc-turing-logo"
              src="turing-full-black.png"
            />
            <div
              className="mc-github-btn"
            >
              <a
                className="mc-gh-tagline"
                href="https://github.com/login/oauth/authorize?client_id=81ef33f7e5b1e21f0162&scope=repo"
              >
                Sign in with GitHub
              </a>
              <img
                alt="GitHub Logo"
                className="mc-github-logo"
                src={
                  Object {
                    "ReactComponent": [Function],
                    "__esModule": true,
                    "default": "github-logo.svg",
                  }
                }
              />
            </div>
          </div>
          <div
            className="mc-title"
          >
            <h1
              className="mc-title-text"
            >
              Mentor Profiles
            </h1>
            <div
              className="mc-line-break"
            />
            <p
              className="mc-tagline"
            >
              Browse current mentors to find your match!
            </p>
          </div>
        </header>,
        <Connect(Search) />,
        <div
          className="mc-card-container"
        />,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <div
              className="mc-nav-btns"
            >
              <img
                alt="Turing Logo"
                className="mc-turing-logo"
                src="turing-full-black.png"
              />
              <div
                className="mc-github-btn"
              >
                <a
                  className="mc-gh-tagline"
                  href="https://github.com/login/oauth/authorize?client_id=81ef33f7e5b1e21f0162&scope=repo"
                >
                  Sign in with GitHub
                </a>
                <img
                  alt="GitHub Logo"
                  className="mc-github-logo"
                  src={
                    Object {
                      "ReactComponent": [Function],
                      "__esModule": true,
                      "default": "github-logo.svg",
                    }
                  }
                />
              </div>
            </div>,
            <div
              className="mc-title"
            >
              <h1
                className="mc-title-text"
              >
                Mentor Profiles
              </h1>
              <div
                className="mc-line-break"
              />
              <p
                className="mc-tagline"
              >
                Browse current mentors to find your match!
              </p>
            </div>,
          ],
          "className": "mc-header",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <img
                  alt="Turing Logo"
                  className="mc-turing-logo"
                  src="turing-full-black.png"
                />,
                <div
                  className="mc-github-btn"
                >
                  <a
                    className="mc-gh-tagline"
                    href="https://github.com/login/oauth/authorize?client_id=81ef33f7e5b1e21f0162&scope=repo"
                  >
                    Sign in with GitHub
                  </a>
                  <img
                    alt="GitHub Logo"
                    className="mc-github-logo"
                    src={
                      Object {
                        "ReactComponent": [Function],
                        "__esModule": true,
                        "default": "github-logo.svg",
                      }
                    }
                  />
                </div>,
              ],
              "className": "mc-nav-btns",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "alt": "Turing Logo",
                  "className": "mc-turing-logo",
                  "src": "turing-full-black.png",
                },
                "ref": null,
                "rendered": null,
                "type": "img",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    <a
                      className="mc-gh-tagline"
                      href="https://github.com/login/oauth/authorize?client_id=81ef33f7e5b1e21f0162&scope=repo"
                    >
                      Sign in with GitHub
                    </a>,
                    <img
                      alt="GitHub Logo"
                      className="mc-github-logo"
                      src={
                        Object {
                          "ReactComponent": [Function],
                          "__esModule": true,
                          "default": "github-logo.svg",
                        }
                      }
                    />,
                  ],
                  "className": "mc-github-btn",
                },
                "ref": null,
                "rendered": Array [
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": "Sign in with GitHub",
                      "className": "mc-gh-tagline",
                      "href": "https://github.com/login/oauth/authorize?client_id=81ef33f7e5b1e21f0162&scope=repo",
                    },
                    "ref": null,
                    "rendered": "Sign in with GitHub",
                    "type": "a",
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "alt": "GitHub Logo",
                      "className": "mc-github-logo",
                      "src": Object {
                        "ReactComponent": [Function],
                        "__esModule": true,
                        "default": "github-logo.svg",
                      },
                    },
                    "ref": null,
                    "rendered": null,
                    "type": "img",
                  },
                ],
                "type": "div",
              },
            ],
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <h1
                  className="mc-title-text"
                >
                  Mentor Profiles
                </h1>,
                <div
                  className="mc-line-break"
                />,
                <p
                  className="mc-tagline"
                >
                  Browse current mentors to find your match!
                </p>,
              ],
              "className": "mc-title",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Mentor Profiles",
                  "className": "mc-title-text",
                },
                "ref": null,
                "rendered": "Mentor Profiles",
                "type": "h1",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "className": "mc-line-break",
                },
                "ref": null,
                "rendered": null,
                "type": "div",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Browse current mentors to find your match!",
                  "className": "mc-tagline",
                },
                "ref": null,
                "rendered": "Browse current mentors to find your match!",
                "type": "p",
              },
            ],
            "type": "div",
          },
        ],
        "type": "header",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [],
          "className": "mc-card-container",
        },
        "ref": null,
        "rendered": Array [],
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <header
            className="mc-header"
          >
            <div
              className="mc-nav-btns"
            >
              <img
                alt="Turing Logo"
                className="mc-turing-logo"
                src="turing-full-black.png"
              />
              <div
                className="mc-github-btn"
              >
                <a
                  className="mc-gh-tagline"
                  href="https://github.com/login/oauth/authorize?client_id=81ef33f7e5b1e21f0162&scope=repo"
                >
                  Sign in with GitHub
                </a>
                <img
                  alt="GitHub Logo"
                  className="mc-github-logo"
                  src={
                    Object {
                      "ReactComponent": [Function],
                      "__esModule": true,
                      "default": "github-logo.svg",
                    }
                  }
                />
              </div>
            </div>
            <div
              className="mc-title"
            >
              <h1
                className="mc-title-text"
              >
                Mentor Profiles
              </h1>
              <div
                className="mc-line-break"
              />
              <p
                className="mc-tagline"
              >
                Browse current mentors to find your match!
              </p>
            </div>
          </header>,
          <Connect(Search) />,
          <div
            className="mc-card-container"
          />,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <div
                className="mc-nav-btns"
              >
                <img
                  alt="Turing Logo"
                  className="mc-turing-logo"
                  src="turing-full-black.png"
                />
                <div
                  className="mc-github-btn"
                >
                  <a
                    className="mc-gh-tagline"
                    href="https://github.com/login/oauth/authorize?client_id=81ef33f7e5b1e21f0162&scope=repo"
                  >
                    Sign in with GitHub
                  </a>
                  <img
                    alt="GitHub Logo"
                    className="mc-github-logo"
                    src={
                      Object {
                        "ReactComponent": [Function],
                        "__esModule": true,
                        "default": "github-logo.svg",
                      }
                    }
                  />
                </div>
              </div>,
              <div
                className="mc-title"
              >
                <h1
                  className="mc-title-text"
                >
                  Mentor Profiles
                </h1>
                <div
                  className="mc-line-break"
                />
                <p
                  className="mc-tagline"
                >
                  Browse current mentors to find your match!
                </p>
              </div>,
            ],
            "className": "mc-header",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <img
                    alt="Turing Logo"
                    className="mc-turing-logo"
                    src="turing-full-black.png"
                  />,
                  <div
                    className="mc-github-btn"
                  >
                    <a
                      className="mc-gh-tagline"
                      href="https://github.com/login/oauth/authorize?client_id=81ef33f7e5b1e21f0162&scope=repo"
                    >
                      Sign in with GitHub
                    </a>
                    <img
                      alt="GitHub Logo"
                      className="mc-github-logo"
                      src={
                        Object {
                          "ReactComponent": [Function],
                          "__esModule": true,
                          "default": "github-logo.svg",
                        }
                      }
                    />
                  </div>,
                ],
                "className": "mc-nav-btns",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "alt": "Turing Logo",
                    "className": "mc-turing-logo",
                    "src": "turing-full-black.png",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": "img",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": Array [
                      <a
                        className="mc-gh-tagline"
                        href="https://github.com/login/oauth/authorize?client_id=81ef33f7e5b1e21f0162&scope=repo"
                      >
                        Sign in with GitHub
                      </a>,
                      <img
                        alt="GitHub Logo"
                        className="mc-github-logo"
                        src={
                          Object {
                            "ReactComponent": [Function],
                            "__esModule": true,
                            "default": "github-logo.svg",
                          }
                        }
                      />,
                    ],
                    "className": "mc-github-btn",
                  },
                  "ref": null,
                  "rendered": Array [
                    Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "host",
                      "props": Object {
                        "children": "Sign in with GitHub",
                        "className": "mc-gh-tagline",
                        "href": "https://github.com/login/oauth/authorize?client_id=81ef33f7e5b1e21f0162&scope=repo",
                      },
                      "ref": null,
                      "rendered": "Sign in with GitHub",
                      "type": "a",
                    },
                    Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "host",
                      "props": Object {
                        "alt": "GitHub Logo",
                        "className": "mc-github-logo",
                        "src": Object {
                          "ReactComponent": [Function],
                          "__esModule": true,
                          "default": "github-logo.svg",
                        },
                      },
                      "ref": null,
                      "rendered": null,
                      "type": "img",
                    },
                  ],
                  "type": "div",
                },
              ],
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <h1
                    className="mc-title-text"
                  >
                    Mentor Profiles
                  </h1>,
                  <div
                    className="mc-line-break"
                  />,
                  <p
                    className="mc-tagline"
                  >
                    Browse current mentors to find your match!
                  </p>,
                ],
                "className": "mc-title",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Mentor Profiles",
                    "className": "mc-title-text",
                  },
                  "ref": null,
                  "rendered": "Mentor Profiles",
                  "type": "h1",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "className": "mc-line-break",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": "div",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Browse current mentors to find your match!",
                    "className": "mc-tagline",
                  },
                  "ref": null,
                  "rendered": "Browse current mentors to find your match!",
                  "type": "p",
                },
              ],
              "type": "div",
            },
          ],
          "type": "header",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [],
            "className": "mc-card-container",
          },
          "ref": null,
          "rendered": Array [],
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
